src/DiagnosticMessages.ts

```diff
@@ -24,711 +28,860 @@ export let DiagnosticMessages = {
+        legacyCode: 1098,
+        severity: DiagnosticSeverity.Error,
+        code: 'child-field-type-not-assignable'
     }),
     classChildMemberDifferentMemberTypeThanAncestor: (memberType: string, parentMemberType: string, parentClassName: string) => ({
         message: `Class member is a ${memberType} here but a ${parentMemberType} in ancestor class '${parentClassName}'`,
-        code: 1099,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1099,
+        severity: DiagnosticSeverity.Error,
+        code: 'child-field-type-different'
     }),
     classConstructorMissingSuperCall: () => ({
         message: `Missing "super()" call in class constructor method.`,
-        code: 1100,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1100,
+        severity: DiagnosticSeverity.Error,
+        code: 'missing-super-call'
     }),
     classConstructorIllegalUseOfMBeforeSuperCall: () => ({
         message: `Illegal use of "m" before calling "super()"`,
-        code: 1101,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1101,
+        severity: DiagnosticSeverity.Error,
+        code: 'use-m-before-super'
     }),
     classFieldCannotBeOverridden: () => ({
         message: `Class field cannot be overridden`,
-        code: 1102,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1102,
+        severity: DiagnosticSeverity.Error,
+        code: 'class-field-override'
     }),
     unusedAnnotation: () => ({
         message: `This annotation is not attached to any statement`,
-        code: 1103,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1103,
+        severity: DiagnosticSeverity.Error,
+        code: 'unused-annotation'
     }),
     localVarShadowedByScopedFunction: () => ({
         message: `Declaring a local variable with same name as scoped function can result in unexpected behavior`,
-        code: 1104,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1104,
+        severity: DiagnosticSeverity.Error,
+        code: 'local-var-shadowed-by-function'
     }),
     scopeFunctionShadowedByBuiltInFunction: () => ({
-        message: `Scope function will not be accessible because it has the same name as a built-in function`,
-        code: 1105,
-        severity: DiagnosticSeverity.Error
+        message: `Scope function will not be accessible because it has the same name as a built -in function`,
+        legacyCode: 1105,
+        severity: DiagnosticSeverity.Error,
+        code: 'function-shadowed-by-built-in-function'
     }),
     localVarSameNameAsClass: (className: string) => ({
         message: `Local variable has same name as class '${className}'`,
-        code: 1106,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1106,
+        severity: DiagnosticSeverity.Error,
+        code: 'local-var-same-name-as-class'
```

Comment:
Since only classes and functions exist at runtime, (and they both result in functions at runtime), we can utilize `localVarShadowedByScopedFunction`.